src/queue.cpp

```diff
@@ -247,6 +249,14 @@ cl_int cvk_command_queue::wait_for_events(cl_uint num_events,
         }
     }
 
+    if (queues_to_flush.size() == 1) {
+        for (auto q : queues_to_flush) {
```

Comment:
Ok, we might want to remove the loop to make this obvious. We could support multi-queue scenarios but it would be a lot more complex. I'll raise an issue.